arm/Microsoft.Compute/virtualMachines/.bicep/nested_networkInterface.bicep

```diff
@@ -67,6 +68,11 @@ resource networkInterface 'Microsoft.Network/networkInterfaces@2021-03-01' = {
       name: !empty(ipConfiguration.name) ? ipConfiguration.name : null
       properties: {
         primary: ((index == 0) ? true : false)
+        applicationSecurityGroups: !empty(applicationSecurityGroupId) ? [
```

Comment:
As this is nested in the ipconfiguration, I wonder if it should be part of this parameter, instead of its own. Otherwise you can only apply the same applicationSecurityGroup(s) to all ipConfigurations.

For example it could be:
```bicep
applicationSecurityGroups: contains(ipConfiguration, 'applicationSecurityGroups') ? ipConfiguration.applicationSecurityGroups : []
```
```

Reply:
@rahalan , thoughts?

Reply:
Further, we should think about adding it as a test case to the linux & windows parameters.json files too and reference the following group: `adp-carml-az-asg-x-001`

Reply:
> 

Ran couple of tests to check ASG settings in multiple scenarios, in general looks like ASG settings apply at NIC level so all ipconfigurations inherit the same ASG settings you implement in one. Only way to have different ASG settings in a VM if by attaching an additional NIC

Reply:
> Further, we should think about adding it as a test case to the linux & windows parameters.json files too and reference the following group: `adp-carml-az-asg-x-001`

Makes sense to add it into the test cases (no idea how to do this one, will need your help)

Reply:
No problem. Let's chat about that